### PR TITLE
docs: add client helpers and e2e in docs

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: -

### Changes included:

- [x] add some missing context in the contributing guide
- [x] add small description about helpers and link to the JS implementation
- [x] mention e2e in docs

### CODE REVIEW

File: website/docs/contributing/add-new-api-client.md [hunk 1]
```diff
@@ -102,3 +111,12 @@ You can find all the commands in the [CLI > clients commands page](/docs/contrib
...
+| ------------- | -------------------------------------------------------------------------------------------------------------|----------------- |-------------------------------------------------------- |-------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `waitForTask` | Given a `taskID`, calls the `getTask` method until the status gets `published`                               | `getTask()`      | `response.status == "published"`                        | [JavaScript](https://github.com/algolia/api-clients-automation/blob/main/clients/algoliasearch-client-javascript/packages/client-search/src/searchClient.ts#L232) |
+| `waitForApi`  | Given a `Key`, calls the `getApiKey` method until the stop condition for the given `operation` is validated  | `getApiKey()`    | Diff between the given `Key` and the `response` payload | [JavaScript](https://github.com/algolia/api-clients-automation/blob/main/clients/algoliasearch-client-javascript/packages/client-search/src/searchClient.ts#L269) |
```
Comment (Fluf22) [inline website/docs/contributing/add-new-api-client.md:122]: waitForApiKey, no?

**Review by Fluf22 — DISMISSED** [review]
Nice additions!

File: website/docs/contributing/add-new-api-client.md (hunk 1 above)
Comment (shortcuts) [reply website/docs/contributing/add-new-api-client.md:122]: ah yes ty!

**Comment by Fluf22** [comment]
Lfg 🚀

**Review by Fluf22 — APPROVED** [review]

**Pull request merged**